src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaProducerAsyncCollector.cs

```diff
@@ -2,19 +2,25 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
 using System.Runtime.CompilerServices;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using System.Timers;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.Output;
 using Newtonsoft.Json.Linq;
 
 [assembly: InternalsVisibleTo("Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
 namespace Microsoft.Azure.WebJobs.Extensions.Kafka
 {
-    internal class KafkaProducerAsyncCollector<T> : IAsyncCollector<T>
+    internal class KafkaProducerAsyncCollector<T> : IAsyncCollector<T>, IDisposable
     {
         private readonly KafkaProducerEntity entity;
         private readonly Guid functionInstanceId;
+        private List<object> eventList = new List<object>();
```

Comment:
Is it thread safe?
use the https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/ or lock to protect it.

Reply:
As per design of webjob sdk every function has it's own IAsyncCollector object instantiation thus I think it's threadsafe